doc-surrealdb_versioned_docs/version-1.x/tutorials/define-a-schema.mdx

```diff
@@ -0,0 +1,230 @@
+
+While this might look like many decisions, you are Defining a Schema.
+
+A Schema defines the structure and organisation of data. It dictates how data is stored, organised, and manipulated. Schemas can specify tables, fields (columns), data types, constraints, and relationships between tables.
+
+In summary, a schema is the primary way to ensure your data acts as expected.
+
+There are two main types of schemas:
+
+1. **Schemafull (Structured)**
+2. **Schemaless (Unstructured)**
+
+When starting a new project in SurrealDB, you can define your Schema using either method depending on your application's requirements. Since SurrealDB is a multi-model database, you can have both in the same project, depending on your needs.
+
+In this tutorial, you will learn how to define a schema in SurrealDB, what using either schema type means for data retrieval, and how to use what you need as your product grows.
+
+## Prerequisites
+
+Before you start, this guide assumes the following:
+
+- Basic knowledge of databases and data modelling.
+- SurrealDB is installed on your machine. You can download and install SurrealDB from the [installation page](/docs/surrealdb/installation/).
+- [A Command line interface (CLI)](/docs/surrealdb/cli/sql) for interacting with SurrealDB or a [Surrealist sandbox](/docs/surrealist/#connecting-to-the-sandbox).
+
+## Schemafull (Structured) Databases
+
+A schemafull database requires the upfront definition of the structure of your data, including collections (tables) and fields (columns). This approach enforces consistency and integrity, making it suitable for applications with well-defined data models.
+
+In surrealDB, the Schemafull approach is usually associated with the [Define statement](/docs/surrealdb/surrealql/statements/define/), which is used to specify instructions to the Schema, such as authentication access and behaviour, global parameters, table configurations, table events, analyzers, and indexes. You can set a schemafull table in the following steps:
+
+### Creating a schemafull table
+
+1. Define a Table: To start a schemafull table, specifically use the Define Table statement.
+
+```surql
+-- Create a schemafull user table.
+DEFINE TABLE user SCHEMAFULL;
+```
+
+2. Define Fields:
+
+```surql
+-- Define some fields.
+DEFINE FIELD firstName ON TABLE user TYPE string;
+DEFINE FIELD lastName ON TABLE user TYPE string;
+DEFINE FIELD email ON TABLE user TYPE string;
+  ASSERT string::is::email($value);
+```
+
+In the code above you may notice the `ASSERT` clause. This can be used to validate any restrictions you want on a field. In the example above the [`string::is::email`](/docs/surrealdb/surrealql/functions/database/string#stringisemail) function is used to check whether the value is an email.
```

Comment:
I think a link to the predefined variables page could work here for the curious reader:

...to check whether [the value](https://surrealdb.com/docs/surrealdb/surrealql/parameters#reserved-variable-names) is an email.